nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/TwitterStreamAPI.java

```
@@ -0,0 +1,174 @@
+        this.pollFields = parseCommaSeparatedListPropreties(context, POLL_FIELDS_PROPERTY_NAME);
+        this.placeFields = parseCommaSeparatedListPropreties(context, PLACE_FIELDS_PROPERTY_NAME);
+        this.expansions = parseCommaSeparatedListPropreties(context, EXPANSIONS_PROPERTY_NAME);
+        this.backfillMinutes = context.getProperty(BACKFILL_MINUTES_PROPERTY_NAME).asInteger();
+
+        TwitterCredentialsBearer creds = new TwitterCredentialsBearer(context.getProperty(BEARER_TOKEN_PROPERTY_NAME).getValue());
+        api = new TwitterApi();
+        api.setTwitterCredentials(creds);
+
+        this.executorService = Executors.newSingleThreadExecutor();
+    }
+
+    public String getBasePath() {
+        return api.getApiClient().getBasePath();
+    }
+
+    public void setBasePath(final String path) {
+        api.getApiClient().setBasePath(path);
+    }
+
+    /**
+     * This method would be called when we would like the stream to get started. This method will spin off a thread that
+     * will continue to queue tweets on to the given queue passed in the constructor. The thread will continue
+     * to run until {@code stop} is called.
+     * @param endpoint {@code TwitterStreamAPI.SAMPLE_ENDPOINT} or {@code TwitterStreamAPI.SEARCH_ENDPOINT}
+     */
+    public void start(final String endpoint) {
+        try {
+            if (endpoint.equals(SAMPLE_ENDPOINT)) {
+                stream = api.tweets().sampleStream(expansions, tweetFields, userFields, mediaFields, placeFields, pollFields, backfillMinutes);
+            } else if (endpoint.equals(SEARCH_ENDPOINT)) {
+                stream = api.tweets().searchStream(expansions, tweetFields, userFields, mediaFields, placeFields, pollFields, backfillMinutes);
+            } else {
+                throw new ProcessException("Endpoint was invalid value: " + endpoint);
+            }
+        } catch (ApiException e) {
+            logger.error("Received error {}: {}", e.getCode(), e.getMessage());
```

Instead of logging an error, the details should be wrapped and thrown in a new `ProcessException`, since processing cannot continue if the initial configuration fails.

---

Will make the changes